src/config.c

```
@@ -2552,6 +2304,207 @@ static int updateTLSPort(long long val, long long prev, const char **err) {
+    UNUSED(data);
+    UNUSED(update);
+    return updateClientOutputBufferLimit(argv, argc, err);
+}
+
+static void getConfigClientOutputBufferLimitOption(client *c, typeData data) {
+    UNUSED(data);
+    sds buf = sdsempty();
+    int j;
+    for (j = 0; j < CLIENT_TYPE_OBUF_COUNT; j++) {
+        buf = sdscatprintf(buf,"%s %llu %llu %ld",
+                           getClientTypeName(j),
+                           server.client_obuf_limits[j].hard_limit_bytes,
+                           server.client_obuf_limits[j].soft_limit_bytes,
+                           (long) server.client_obuf_limits[j].soft_limit_seconds);
+        if (j != CLIENT_TYPE_OBUF_COUNT-1)
+            buf = sdscatlen(buf," ",1);
+    }
+    addReplyBulkCString(c,buf);
+    sdsfree(buf);
+}
+
+static int setConfigOOMScoreAdjValuesOption(typeData data, sds *argv, int argc, int update, const char **err) {
+    UNUSED(data);
+    if (argc != CONFIG_OOM_COUNT) {
+        *err = "wrong number of arguments";
+        return 0;
+    }
+    if (updateOOMScoreAdjValues(argv,err,update) == C_ERR) return 0;
+    return 1;
+}
+
+static void getConfigOOMScoreAdjValuesOption(client *c, typeData data) {
+    UNUSED(data);
+    sds buf = sdsempty();
+    int j;
+
+    for (j = 0; j < CONFIG_OOM_COUNT; j++) {
+        buf = sdscatprintf(buf,"%d", server.oom_score_adj_values[j]);
+        if (j != CONFIG_OOM_COUNT-1)
+            buf = sdscatlen(buf," ",1);
+    }
+
+    addReplyBulkCString(c,buf);
+    sdsfree(buf);
+}
+
+static int setConfigNotifyKeyspaceEventsOption(typeData data, sds *argv, int argc, int update, const char **err) {
+    UNUSED(data);
+    if (argc != 1) {
+        *err = "wrong number of arguments";
+        return 0;
+    }
+    int flags = keyspaceEventsStringToFlags(argv[0]);
+    if (flags == -1) {
+        if (update) *err = "Invalid event class character. Use 'g$lshzxeA'.";
```

why not return a better error when `!update` too?
should i conclude from this diff that whoever created these common "set" functions was responsible enough to make a diff between the config file parsing handling and the config set and make sure they're either exactly the same, or tweak the code as needed? (i.e. not just copying one block and deleting the other blindly).

in this specific case i think the difference between the two blocks (the error response) was not something we wanna preserve...

another thing that concerns me, it looks like the example is outdated, in unstable i see `Use 'Ag$lshzxeKEtmd'`
so maybe this project started on an old version and then conflicts where not properly resolved when rebased?

---

> should i conclude from this diff that whoever created these common "set" functions was responsible enough to make a diff between the config file parsing handling and the config set and make sure they're either exactly the same, or tweak the code as needed? (i.e. not just copying one block and deleting the other blindly).
> 
Yes that what Bjorn did. But I think he reversed the logic. The original code printed the details when parsing the config file and now we print it on `CONFIG SET`. Anyway We should print it in both cases I think so I'll fix it.

> another thing that concerns me, it looks like the example is outdated, in unstable i see `Use 'Ag$lshzxeKEtmd'` so maybe this project started on an old version and then conflicts where not properly resolved when rebased?

I guess I missed it in the rebase. I'll fix.